### PR TITLE
DOC: Release notes for v1.0.1.

### PR DESCRIPTION
**Pull request by hhslepicka**
## Description

Release notes in preparation for v1.0.1 release.

## Motivation and Context
Prepare for new release.

## How Has This Been Tested?
Locally.

## Where Has This Been Documented?
It is documentation.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -2,6 +2,28 @@
  Release History
 =================
 
+
+v1.0.1 (2020-05-29)
```
Comment (klauer) [inline docs/source/release_notes.rst:6]:
```suggestion
v1.0.1 (2020-05-20)
```

**Comment by klauer** [comment]
Did you build locally? CI isn't building docs until merged (#334)

**Comment by hhslepicka** [comment]
@ZLLentz I moved it to fixes.
@klauer docs are building locally with no warnings.
```$ make clean html
rm -rf build
rm -rf source/generated
Running Sphinx v2.2.2
Questionnaire backend unavailable: No module named 'psdm_qs_cli'
making output directory... done
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
[autosummary] generating autosummary for: basic_usage.rst, cli.rst, connections.rst, display.rst, index.rst, plugins.rst, python_methods.rst, release_notes.rst, save.rst, tools.rst, utils.rst, widgets.rst
[autosummary] generating autosummary for: /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosConsole.rst, /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosLogDisplay.rst, /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosTimePlot.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 12 source files that are out of date
updating environment: [new config] 15 added, 0 changed, 0 removed
/Users/slepicka/mc/envs/py37/lib/python3.7/site-packages/sphinx/util/nodes.py:151: FutureWarning:
   The iterable returned by Node.traverse()
   will become an iterator instead of a list in Docutils > 0.16.
  for classifier in reversed(node.parent.traverse(nodes.classifier)):
reading sources... [100%] widgets
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] widgets
generating indices...  genindex py-modindexdone
highlighting module code... [100%] typhos.widgets
writing additional pages...  search/Users/slepicka/mc/envs/py37/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {{ super() }}
done
copying images... [100%] _static/expanded.jpg
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in build/html.```

**Comment by klauer** [comment]
Sphinx is up to 3 now and may give different warnings. Here goes nothing...

**Comment by hhslepicka** [comment]
@klauer just updated... still clean...

```
$ make clean html
rm -rf build
rm -rf source/generated
Running Sphinx v3.0.3
Questionnaire backend unavailable: No module named 'psdm_qs_cli'
making output directory... done
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
[autosummary] generating autosummary for: basic_usage.rst, cli.rst, connections.rst, display.rst, index.rst, plugins.rst, python_methods.rst, release_notes.rst, save.rst, tools.rst, utils.rst, widgets.rst
[autosummary] generating autosummary for: /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosConsole.rst, /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosLogDisplay.rst, /Users/slepicka/sandbox/git-pcdshub/typhon-git/docs/source/generated/typhos.tools.TyphosTimePlot.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 12 source files that are out of date
updating environment: [new config] 15 added, 0 changed, 0 removed
reading sources... [100%] widgets
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] widgets
generating indices...  genindex py-modindexdone
highlighting module code... [100%] typhos.widgets
writing additional pages...  searchdone
copying images... [100%] _static/expanded.jpg
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in build/html.
```

**Pull request merged**